tooling/lsp/src/requests/mod.rs

```diff
@@ -217,29 +258,36 @@ pub(crate) fn on_initialize(
         .unwrap_or_default();
     state.options = initialization_options;
 
-    async move {
-        let text_document_sync = TextDocumentSyncCapability::Kind(TextDocumentSyncKind::FULL);
-
-        let code_lens = if initialization_options.enable_code_lens {
-            Some(CodeLensOptions { resolve_provider: Some(false) })
-        } else {
-            None
-        };
-
-        let nargo = NargoCapability {
-            tests: Some(NargoTestsOptions {
-                fetch: Some(true),
-                run: Some(true),
-                update: Some(true),
-            }),
-        };
+    let enable_code_lens =
+        !initialization_options.enable_lightweight_mode && initialization_options.enable_code_lens;
+    let enable_inlay_hints = !initialization_options.enable_lightweight_mode
+        && initialization_options.enable_inlay_hints;
+    let enable_completions = !initialization_options.enable_lightweight_mode
+        && initialization_options.enable_completions;
+    let enable_signature_help = !initialization_options.enable_lightweight_mode
+        && initialization_options.enable_signature_help;
+    let enable_code_actions = !initialization_options.enable_lightweight_mode
+        && initialization_options.enable_code_actions;
 
+    async move {
         Ok(InitializeResult {
             capabilities: ServerCapabilities {
-                text_document_sync: Some(text_document_sync),
-                code_lens_provider: code_lens,
+                text_document_sync: Some(TextDocumentSyncCapability::Kind(
+                    TextDocumentSyncKind::FULL,
+                )),
+                code_lens_provider: if enable_code_lens {
+                    Some(CodeLensOptions { resolve_provider: Some(false) })
+                } else {
+                    None
+                },
```

Comment:
```suggestion
                code_lens_provider: enable_code_lens.then_some(
                    CodeLensOptions { resolve_provider: Some(false) }
                ),
```
:nerd_face:

Reply:
Aaah... I was sure there was a bool method for this but I thought it would end up looking bad, but it's actually better! I'll make this change.

Reply:
Done!